autopush/router/gcm.py

```diff
@@ -34,10 +34,27 @@ def __init__(self, ap_settings, router_conf):
             raise IOError("GCM Bridge not initiated in main")
         self.log.debug("Starting GCM router...")
 
-    def check_token(self, token):
+    def check_token(self, token, relo):
         if token not in self.senderIDs.senderIDs():
-            return (False, self.senderIDs.choose_ID().get('senderID'))
-        return (True, token)
+            # It is possible for some systems to allow the token to be
+            # changed, or a new token to be sent. At least one system
+            # historically allowed this. Unfortunately, that system no
+            # longer does. Currently, all mobile systems treat non
+            # 200 as an error.
+            # Until we support a system that allows for relocation, we
+            # are codifying to return a "400", which is technically both
+            # correct, and incorrect, depending on your point of view.
+            #
+            # if relocation were enabled...
+            #
+            # token = self.senderIDs.choose_ID().get('senderID')
+            # relo_url = relo % token
+            # return writer(301, 0 "Location: %s" + relo_url,
+            #               headers={"Location": relo_url)
+            return False, dict(status_code=400,
```

Comment:
I'm wondering if this could throw a `RouterException` instead, and then this method wouldn't need to return anything. Or is that too magical?